src/System/Taffybar.hs

```diff
@@ -175,12 +182,15 @@ data TaffybarConfig =
                  , endWidgets :: [IO Widget] -- ^ Widgets that are packed from right-to-left in the bar
                  }
 
+type TaffybarConfigEQ = (TaffybarConfig, StableName TaffybarConfig)
```

Comment:
The reason for using stablenames is so that it is possible to change the taffybar config dynamically for a particular bar at any time. You don't want to completely recreate the bar if nothing has changed, and normally you could just use (==) for this, but because taffybar has fields that are functions an IO operations, this is not possible.

There are definitely other ways to handle this, but this felt like the easiest way to handle this to me.